src/crypto/CHIPCryptoPAL.h

```diff
@@ -99,15 +99,45 @@ enum class SupportedECPKeyTypes : uint8_t
     ECP256R1 = 0,
 };
 
+template <typename Sig>
 class ECPKey
 {
 public:
     virtual SupportedECPKeyTypes Type() const = 0;
     virtual size_t Length() const             = 0;
     virtual operator uint8_t *() const        = 0;
+
+    virtual CHIP_ERROR ECDSA_validate_msg_signature(const uint8_t * msg, const size_t msg_length, const Sig & signature) const
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+};
+
+class P256ECDSASignature
+{
+public:
+    size_t & Length() { return length; }
+    size_t Length() const { return length; }
+    operator uint8_t *() const { return (uint8_t *) bytes; }
```

Comment:
Should we have a .data() here instead of implicit conversion? Static typing is less useful if everything is implicitly convertible to bytes.

Reply:
I like this conversion, as it allows us to do something like this. And, this is also the main use of this structure. So flow of the code looks simpler.

```
P256ECDSASignature sig;
int result = EVP_DigestSignFinal(context, Uint8::to_uchar(sig), sig.Length());
```

Reply:
Maybe rename to LengthRef() instead?

When I look at that piece of code, it is not easy to parse for me.
Alternatively maybe we could make the class call digest rather than digest call the class.

Reply:
@andy31415 , this conversion pattern was suggested as a review comment in another one of my PRs. Let me see what can be done to cleanly resolve it. It'll probably go in a separate PR, though.